ConfigurationSystem/Client/Helpers/Resources.py

```diff
@@ -89,14 +89,18 @@ def getFTS3Servers():
   return S_OK( ftsServers )
 
 
-def getFTS3ServerName(serverUrl):
-  """ returns the fts server name for a given fts server url
+def getFTS3ServerDict():
+  """ returns a dict of key = server name and value = server url
   """
+
   csPath = cfgPath(gBaseResourcesSection, "FTSEndpoints/FTS3")
-  for resourceName in gConfig.getOptions(csPath).get('Value', []):
-    if serverUrl == gConfig.getValue(cfgPath(csPath, resourceName)):
-      return S_OK(resourceName)
-  return S_ERROR("FTS server not found ", serverUrl)
+  ftsServerNames = gConfig.getOptions(csPath).get('Value', [])
```

Comment:
you could use `getOptionsDict` to do it in one go no ? Or do I misunderstand what you are doing ?

Reply:
good point! getOptionsDict is better..